src/lib/util/Security/ModerationActions.ts

```diff
@@ -447,6 +471,36 @@ export class ModerationActions {
 		return this.removeStickyRole(moderatorID, id, mutedRole);
 	}
 
+	private async nicknameUser(rawOptions: ModerationActionOptions, nickname: string) {
+		try {
+			const member = await this.guild.members.fetch(rawOptions.user_id);
+			return await member.setNickname(nickname);
```

Comment:
Please change this to do a single direct REST call to add a role, instead of fetching and then add the role.